Code/Framework/AzCore/Tests/Math/MathUtilsTests.cpp

```diff
@@ -58,4 +58,55 @@ namespace UnitTest
         EXPECT_NEAR(0.6, AZ::LerpInverse(1.0, 1.0 + 5.0 * epsilonD, 1.0 + 3.0 * epsilonD), epsilonD);
         EXPECT_NEAR(1.0, AZ::LerpInverse(1.0, 1.0 + 5.0 * epsilonD, 1.0 + 5.0 * epsilonD), epsilonD);
     }
+
+    class RoundUpToMultipleTestsFixture
+        : public ScopedAllocatorSetupFixture
+        , public ::testing::WithParamInterface<uint32_t>
+    {
+    };
+
+    template <typename T>
+    void TestRoundUpToMultipleIsCorrect(T alignment)
+    {
+        // Example: alignment: 4
+        // inputValue:     0 1 2 3 4 5 6 7 8 ...
+        // expectedOutput: 0 4 4 4 4 8 8 8 8 ...
+        AZStd::vector<T> expectedOutput;
+        constexpr T iterations = 4;
+        expectedOutput.reserve(iterations * alignment + 1);
+        expectedOutput.push_back(0);
+
+        for (T i = 1; i <= iterations; ++i)
+        {
+            for (T j = 0; j < alignment; ++j)
+            {
+                expectedOutput.push_back(i * alignment);
+            }
+        }
+
+        for (T inputValue = 0; inputValue < expectedOutput.size(); ++inputValue)
+        {
+            T result = RoundUpToMultiple(inputValue, alignment);
+            EXPECT_EQ(result, expectedOutput[inputValue]);
+        }
+    }
+
+    TEST_P(RoundUpToMultipleTestsFixture, RoundUpToMultipleUInt32_ValidInput_IsCorrect)
+    {
+        uint32_t alignment = GetParam();
+        TestRoundUpToMultipleIsCorrect(alignment);
+    }
+
+    TEST_P(RoundUpToMultipleTestsFixture, RoundUpToMultipleUInt64_ValidInput_IsCorrect)
+    {
+        uint64_t alignment = GetParam();
+        TestRoundUpToMultipleIsCorrect(alignment);
+    }
+    
+    INSTANTIATE_TEST_CASE_P(
+        MATH_RoundUpToMultiple,
+        RoundUpToMultipleTestsFixture,
+        // Multiples that we're going to test rounding up to
+        // Test with some low numbers, prime numbers, power of two numbers, and non-prime non-power-of-two numbers
+        ::testing::ValuesIn({1u,2u,3u,4u,5u,8u,9u,12u,13u}));
```

Comment:
I understand that you are trying to test a wide range of data without lots of code, but this doesn't necessarily make for the best unit test. I think it's best practice to avoid logic and loops in unit tests. While it's good for _production code_ to be well-factored and general-purpose, doing that in unit tests makes code hard to read and could introduce bugs in your unit test that make it ineffective. Unit tests should be easy to read and unlikely to have bugs. Consider writing your unit tests like this instead:
```
EXPECT_EQ(RoundUpToMultiple(0, 1), 0);
EXPECT_EQ(RoundUpToMultiple(1, 1), 1);
EXPECT_EQ(RoundUpToMultiple(2, 1), 1);

EXPECT_EQ(RoundUpToMultiple(0, 2), 0);
EXPECT_EQ(RoundUpToMultiple(1, 2), 2);
EXPECT_EQ(RoundUpToMultiple(2, 2), 2);
EXPECT_EQ(RoundUpToMultiple(3, 2), 4);
EXPECT_EQ(RoundUpToMultiple(4, 2), 4);
EXPECT_EQ(RoundUpToMultiple(5, 2), 6);

EXPECT_EQ(RoundUpToMultiple(0, 13), 0);
EXPECT_EQ(RoundUpToMultiple(1, 13), 13);
EXPECT_EQ(RoundUpToMultiple(9, 13), 13);
EXPECT_EQ(RoundUpToMultiple(12, 13), 13);
EXPECT_EQ(RoundUpToMultiple(13, 13), 13);
EXPECT_EQ(RoundUpToMultiple(14, 13), 26);
EXPECT_EQ(RoundUpToMultiple(25, 13), 26);
EXPECT_EQ(RoundUpToMultiple(26, 13), 26);
EXPECT_EQ(RoundUpToMultiple(27, 13), 39);
```
Even though this doesn't cover as many values, IMO it's a better test because of its readability, low chance of unit test bugs, and even serves as documentation for the function.

Reply:
It's also easier to validate this ^ though code review than the loops you have above

Reply:
you can use a Parameterized test.